elliottlib/errata.py

```diff
@@ -414,7 +414,7 @@ def parse_exception_error_message(e):
     return [int(b.split('#')[1]) for b in re.findall(r'Bug #[0-9]*', str(e))]
 
 
-def add_bugs_with_retry(advisory, bugs, noop=False, batch_size=100):
+def add_bugs_with_retry(advisory, bugs, bzapi, noop=False, batch_size=100):
```

Comment:
I would prefer to filter bugs that are already attached to an advisory out before calling `add_bugs_with_retry` rather than filter in `add_bugs_with_retry`. That will decouple the BZ api calls from errata module and keep `add_bugs_with_retry` simpler. Thoughts?

Reply:
Another reason to do the filtering out of this function is that `add_bugs_with_retry` should sound what it is like. If it fails to attach specified bugs to specific advisory, it should raise an exception so we know there's an error. If we have the filter in this function, it will just return without any error even if the already attached bugs are on a different advisory, which looks like an unexpected behavior to me.

Reply:
+1 I have refactored and moved this to another function